sdks/java/extensions/euphoria/euphoria-beam/src/main/java/org/apache/beam/sdk/extensions/euphoria/beam/JoinTranslator.java

```diff
@@ -93,7 +93,7 @@
     return coGrouped.apply(joinFn.getFnName(), ParDo.of(joinFn));
   }
 
-  private <K, ValueT> PCollection<KV<K, ValueT>> getKVInputCollection(
```

Comment:
At this point the `getKVInputCollection` method has become utility shared among classes. Please consider pulling the method out to some separate utility class to emphasis reusability.